.github/workflows/build_and_test.yml

```diff
@@ -15,6 +15,9 @@ jobs:
     - name: Check out repo
       uses: actions/checkout@v2
 
+    - name: Clone dependency packages
+      run: mkdir dependency_ws && vcs import dependency_ws < build_depends.repos
+
     - name: Install missing dependencies
       run: rosdep update && DEBIAN_FRONTEND=noninteractive sudo rosdep install --from-paths . --ignore-src --rosdistro foxy -y
```

Comment:
Can you add `apt-get update &&` before `rosdep update`? This should fix the issue of APT not finding xacro

Reply:
Fixed.